src/lib/storage/index/abstract_table_index.hpp

```diff
@@ -79,19 +79,70 @@ class AbstractTableIndex : private Noncopyable {
   AbstractTableIndex(AbstractTableIndex&&) = delete;
   virtual ~AbstractTableIndex() = default;
 
-  // For iterating from _cbegin() to _cend()
+  /**
+   * The following four methods are used to access any table index. Each of them accepts a generic lambda (or similar)
```

Comment:
Function object instead of lambda to be more general since we don't have this restriction?